plugins/tracers/gstqueuelevel.c

```diff
@@ -61,6 +61,29 @@ static const gchar queue_level_metadata_event[] = "event {\n\
 };\n\
 \n";
 
+static GstElement *
+get_parent_element (GstPad * pad)
+{
+  GstElement *element = NULL;
+  GstObject *parent;
+  GstObject *child = GST_OBJECT (pad);
+
+  do {
+    parent = gst_object_get_parent (child);
```

Comment:
Note the difference between `gst_object_get_parent` and `GST_OBJECT_PARENT`. The first one gives you a reference, so you need to free it afterwards. The second is faster, and does not give you a reference. In general this is GStreamer standard, `lower_case_functions` are transfer full, and `UPPER_CASE_MACROS` are transfer none.

Lets use the second one.